force-app/main/default/classes/GiftBatch.cls

```diff
@@ -211,4 +260,22 @@ public inherited sharing class GiftBatch implements IGiftBatch {
         }
         set;
     }
+
+    private Boolean hasFinishedProcessing() {
+        return this.asyncApexJob.Status == UTIL_BatchJobService.ApexJobStatus.ABORTED.name()
+            || this.asyncApexJob.Status == UTIL_BatchJobService.ApexJobStatus.FAILED.name()
+            || this.asyncApexJob.Status == UTIL_BatchJobService.ApexJobStatus.COMPLETED.name();
+    }
+
+    private Boolean hasBeenProcessed() {
+        return this.batch?.Latest_Apex_Job_Id__c != null;
+    }
+
+    private AsyncApexJobId lastProcessedJobId() {
```

Comment:
What happens if the user does not have field level access to this field?

Reply:
Since our permission set work is for 238, we would have to inform customers to grant read and write access to this field on the `DataImportBatch__c` object, then add it to our permission set before 238 gets released.